### DISCUSSION TITLE
Question: How to enable plugins of extended theme?

### DISCUSSION THREAD

**Discussion by joapuiib** [mkdocs/mkdocs#3767] [Q&A] [answered]
Hi,
I don't know if this is the best place to ask a question, but I couldn't find any information related to this in the documentation.

I'm trying to create a custom theme that __extends__ another theme (in this case, I'm extending `mkdocs-material`). My custom theme is installed via pip and placed in a subfolder named __custom_theme__. The basic project structure that I'm using can be found at https://github.com/joapuiib/mkdocs-custom-theme/tree/main.

This theme extends `material` theme, defined in `custom_theme/mkdocs_theme.yml`:
```yml
extends: material
```

It works so far, but I've not been able to enable the built-in `search` plugin available in `mkdocs-material`.

When using `theme: custom-theme`, the search bar is not shown.
```
site_name: My Docs

theme:
  name: custom-theme

plugins:
  - search
```

If I use the `theme: material`, it is displayed.
```
site_name: My Docs

theme:
  name: material

plugins:
  - search
```

## Question
How can I enable plugins defined in extended themes when creating a custom theme with `extends:`?

## What I've tried

One thing I've tried is to redefine the theme plugin entrypoints in `pyproject.toml`, but it does not seem to be working.

```yml
[project.entry-points."mkdocs.plugins"]
"custom_theme/blog" = "material.plugins.blog.plugin:BlogPlugin"
"custom_theme/group" = "material.plugins.group.plugin:GroupPlugin"
"custom_theme/info" = "material.plugins.info.plugin:InfoPlugin"
"custom_theme/offline" = "material.plugins.offline.plugin:OfflinePlugin"
"custom_theme/privacy" = "material.plugins.privacy.plugin:PrivacyPlugin"
"custom_theme/search" = "material.plugins.search.plugin:SearchPlugin"
"custom_theme/social" = "material.plugins.social.plugin:SocialPlugin"
"custom_theme/tags" = "material.plugins.tags.plugin:TagsPlugin"
```

Thanks in advance.

**Comment by pawamoy** [comment, marked as answer]
Try using either one of these:

```yaml
plugins:
  - material/search
```

```yaml
plugins:
  - custom_theme/search
```

**Reply by joapuiib** [reply]
Thanks for your fast answer! Works like a charm.
